### ISSUE TITLE
Actor format is incorrect for 'received grade for' event

### ISSUE THREAD

**Issue by dasu2013** [davidpesce/moodle-logstore_xapi#133]
**Description**
- Actor format is incorrect for ‘received grade for’ event. It appears like below in learning locker.
"actor": {
      "account": {
        "homePage": "https://example.com",
        "name": "12345"
      },
      "name": "Ann Smith",
      "objectType": "Agent"
    }
Expected actor format is,
"actor": {
      "mbox": "mailto:12345@example.com",
      "name": " Ann Smith",
      "objectType": "Agent"
    }

**Comment by ryasmi** [comment]
Hi @dasu2013, thanks for reporting your issue.

**Comment by garemoko** [comment]
Here's the relevant bits of code: 

https://github.com/LearningLocker/Moodle-Log-Expander/blob/master/src/Events/AssignmentGraded.php#L18

https://github.com/LearningLocker/Moodle-Log-Expander/blob/86384c3737870e652e924b3aef979c932b23b5cb/src/Repository.php#L234-L253

https://github.com/LearningLocker/Moodle-xAPI-Translator/blob/master/src/Events/AssignmentGraded.php#L36-L38

https://github.com/LearningLocker/xAPI-Recipe-Emitter/blob/master/src/Events/AssignmentGraded.php#L42-L50

The solution is to use this function in the emitter instead of building the actor directly: https://github.com/LearningLocker/xAPI-Recipe-Emitter/blob/master/src/Events/Event.php#L38-L53

**Comment by garemoko** [comment]
ie. in https://github.com/LearningLocker/xAPI-Recipe-Emitter/blob/master/src/Events/AssignmentGraded.php#L42-L50

```
$statement['actor'] = $this->readUser($opts, "graded_user");
```

Gotta love fixes that actually reduce lines of code.

**Comment by Dasuntha** [comment]
Hi Andrew,

Thanks for this reply. The problem is $opts array doesn't have graded_user's email. It has only graded_user_id, graded_user_url and graded_user_name. There is no 'graded_user_email'. Therefore this proposed solution didn't work.

**Comment by Dasuntha** [comment]
Solution for this is
https://github.com/LearningLocker/xAPI-Recipe-Emitter/blob/master/src/Events/AssignmentGraded.php#L42-L50
$statement['actor'] = $this->readUser($opts, "graded_user");

AND

https://github.com/LearningLocker/Moodle-xAPI-Translator/blob/master/src/Events/AssignmentGraded.php#L36-L38

Add user email address into this array,
 return [array_merge(parent::read($opts)[0], [
            'recipe' => 'assignment_graded',
            'graded_user_id' => $opts['graded_user']->id,
            'graded_user_url' => $opts['graded_user']->url,
            'graded_user_name' => $opts['graded_user']->fullname,
            **'graded_user_email' => $opts['graded_user']->email,**
            'grade_score_raw' => $scoreRaw,
            'grade_score_min' => $scoreMin,
            'grade_score_max' => $scoreMax,
            'grade_score_scaled' => $scoreScaled,
            'grade_success' => $success,
            'grade_completed' => true,
            'grade_comment' => strip_tags($opts['grade_comment']),
        ])];

**Comment by ryasmi** [comment]
Thanks @dasu2013 👍

**Comment by davidpesce** [comment]
This was merged with #139
